src/ContinousLatencyBufferModel.hpp

```diff
@@ -21,93 +21,71 @@
-                              std::function<RawType*(unsigned)>& front_override,
-                              std::function<void()>& lock_override,
-                              std::function<void()>& unlock_override) // NOLINT
-//  : folly::ProducerConsumerQueue<RawType>(qsize)
-  : AccessableProducerConsumerQueue<RawType>(qsize)
-  , occupancy_override_(occupancy_override)
-  , write_override_(write_override)
-  , read_override_(read_override)
-  , pop_override_(pop_override)
-  , front_override_(front_override)
-  , lock_override_(lock_override)
-  , unlock_override_(unlock_override)
 
-  {
-    // Bind custom post process of data
-    occupancy_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::occupancy_of_buffer, this);
-    write_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::write_data_to_buffer, this, std::placeholders::_1);
-    read_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::read_data_from_buffer, this, std::placeholders::_1);
-    pop_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::pop_data_from_buffer, this, std::placeholders::_1);
-    front_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::front_data_of_buffer, this, std::placeholders::_1);
-    lock_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::lock_buffer, this);
-    unlock_override_ = std::bind(&ContinousLatencyBufferModel<RawType>::unlock_buffer, this);
+  ContinousLatencyBufferModel() : m_queue(new AccessableProducerConsumerQueue<RawType>(uninitialized_buffer_size)) {
+    TLOG(TLVL_WORK_STEPS) << "Creating uninitialized latency buffer";
   }
 
-  size_t occupancy_of_buffer() {
-    return this->sizeGuess();
+  void conf(const nlohmann::json& cfg) override {
+    auto params = cfg.get<datalinkhandler::Conf>();
+    m_queue.reset(new AccessableProducerConsumerQueue<RawType>(params.latency_buffer_size));
   }
 
-  void lock_buffer() {
-   this->lock();
+  size_t occupancy() override {
+    return m_queue->sizeGuess();
   }
 
-  void unlock_buffer() {
-   this->unlock();
+  void lock() override {
+    m_queue->lock();
+  }
+
+  void unlock() override {
+    m_queue->unlock();
   }
 
   // For the continous buffer, the data is moved into the Folly queue.
   bool
-  write_data_to_buffer(RawType new_element) 
+  write(RawType new_element) override
   {
-    return this->write( std::move(new_element) );
+    return m_queue->write( std::move(new_element) );
   }
 
   bool 
-  read_data_from_buffer(RawType& element) 
+  read(RawType& element) override
   {
-    return this->read(element);
+    return m_queue->read(element);
   }
 
   void 
-  pop_data_from_buffer(unsigned num) // NOLINT
+  pop(unsigned num) override// NOLINT
   {
     for (unsigned i=0; i<num; ++i) { // NOLINT
-      this->popFront();
+      m_queue->popFront();
     }
   }
 
   RawType* 
-  front_data_of_buffer(unsigned idx) // NOLINT
+  front(unsigned idx) override// NOLINT
```

Comment:
Use brackets around if/else blocks please.

I would rename this in the interface:
```
getPtr(unsigned idx = 0 ) {
  if (idx ==0 ) {
    return m_queue->frontPtr();
  } else {
    return m_queue->readPtr(idx);
  }
}
```